mitodl/ocw-studio
# move resource description to markdown body

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of https://github.com/mitodl/ocw-hugo-projects/issues/185

#### What's this PR do?
In https://github.com/mitodl/ocw-hugo-projects/pull/187 we will be changing the `ocw-course` starter to store the description of resources in the markdown body.  This is being done so that linking to other pages and resources inside the description can be supported.  This PR adds a migration that moves existing values in `metadata["description"]` to the markdown body of resources.

#### How should this be manually tested?
 - Take a backup of the production database and restore it to your local instance using:
```
pg_dump PROD_DB_URL > ./dumps/prod_TODAY_DATE.sql
docker-compose rm -sv db
docker-compose up -d db
docker-compose exec -T -e PGPASSWORD=postgres db psql -h db -U postgres < ./dumps/prod_TODAY_DATE.sql
```
 - Spin up your local instance of `ocw-studio` using this backed up database on this branch.  The migrations should run without issue
 - Run `docker-compose run --rm web ./manage.py shell` and run the following in a shell:
```
In [1]: from websites.models import *

In [2]: course_starter = WebsiteStarter.objects.get(name="ocw-course")

In [5]: WebsiteContent.objects.filter(website__starter=course_starter, type="resource", metadata__description__ne="").count()
Out[5]: 0
```
 - You should see an output of 0 indicating all descriptions have been copied over.  The rendering of these changes will be tested in an upcoming `ocw-hugo-themes` PR.
